src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -297,24 +297,10 @@ public string ServiceName
             }
         }
 
-        internal static bool ValidServiceName(string serviceName)
-        {
-            if (serviceName == null)
-                return false;
-
-            // not too long and check for empty name as well.
-            if (serviceName.Length > ServiceBase.MaxNameLength || serviceName.Length == 0)
-                return false;
-
-            // no slashes or backslash allowed
-            foreach (char c in serviceName)
-            {
-                if (c == '\\' || c == '/')
-                    return false;
-            }
-
-            return true;
-        }
+        internal static bool ValidServiceName(string serviceName) =>
```

Comment:
This library builds downlevel. Is it going to need ifdef?

Reply:
I was missing an AsSpan which I fixed, but the library already uses AsSpan downlevel, and MemoryExtensions.IndexOfAny is available in that same downlevel package, so I believe it should be fine.

Reply:
This project references System.Diagnostics.EventLog which references System.Security.Principal.Windows which references System.Memory... presumably it's brought in via that chain of ProjectReferences.